### PR TITLE
Migrate remaining Unboxed repo modules into courses

### PR DESCRIPTION
**Pull request by mikemaccana**
- ~Fix ContentLayer warnings https://github.com/contentlayerdev/contentlayer/issues/495~ this ended up being done in a separate PR.
- Moved course-structure from https://github.com/Unboxed-Software/solana-course/pull/384 to this repo using metadata.yml files
  - The "extending-solana-functionality" section was split into "connecting to offchain data" course (with oracles and vrf) and a separate "solana pay" single-lesson course.
  - 'versioned-transaction' was renamed to 'lookup-tables' (this content is focused on overcoming size limits by lookup tables rather than on versioned transactions per se)
  - 'front end for native' lessons were migrated into the 'native development' course
  - 'env-variables' was renamed to 'program configuration' (as the content is focused on multiple methods for program configuration)
  - 'cpi' and 'vrf' renamed to 'cross-program-invocations' and 'verifiable-randomess-functions' to match the naming convention of the other lessons
  - Tweak the Anchor CPI text to avoid assuming people have read the native CPI lesson.
  - Re-add burning and delegation exercises in a new lesson and lab
- Prioritize courses, so simpler ones are shown first
- Fix internal relative links to the other lessons
- Added images and fixed image links
- Changed any link text that includes '//' to work around an apparent bug in https://github.com/remarkjs/remark-gfm
- Add slashes to escape `<` and `{` in Markdown tables that stop the markdown parser. 
- Add descriptions for all lessons (and ensure descriptions fit in UI for solana.com)
- Use Callouts for tips
- Remove a few remaining instances of old airtable form

### CODE REVIEW

**Comment by mikemaccana** [comment]
Getting:

> `Error: Static page generation for /en/environment is still timing out after 3 attempts. See more info here https://nextjs.org/docs/messages/static-page-generation-timeout` 

- which doesn't seem related to this work.

**Comment by mikemaccana** [comment]
Gettimg:

> Build error occurred
> Error: Static page generation for /en/environment is still timing out after 3 attempts. See more info here https://nextjs.org/docs/messages/static-page-generation-timeout

Which doesn't seem related to this work 🤔

File: content/courses/native-development/metadata.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+author: unboxed
+title: Native onchain program development
```
Comment (nickfrosty) [inline content/courses/native-development/metadata.yml:2]: can we update the slug for this course to be `native-onchain-development`

**Review by nickfrosty — CHANGES_REQUESTED** [review]
I rebased with master already for this PR :) and fixed a lingering import from contentlayer1

I also left a few assorted typo comments throughout. But I stopped looking for those since there are lots.... I was just flagged the ones that my editor would flag to me. We should fix those

some of the code blocks are REALLY big. I think we should use the `details` approach that John Liu recently started using with some of his content PRs (this can be seperate PRs to help tidy things later)

Important fixes right now:
- update all relative links (i.e. `./file`). they do not work in this repo
- the "Onchain program development" course seems wholly focused on Anchor, but it is not clear from the metadata
- the `security-intro` lesson has no heading on the page's content, so it looks really odd on solana-com. can you add some headings that make sense?
- for the form link displayed at the end, you could wrap it in a "success" Callout to make it stand out some more. but if not, that's cool too

File: content/courses/intro-to-solana/intro-to-custom-onchain-programs.md [hunk 2]
```diff
@@ -111,7 +110,7 @@ its data in a specific account at the address
 ### Basic scaffolding
 
 We'll start by using the same packages and `.env` file we made earlier in
-[Intro to Writing Data](/content/courses/intro-to-solana/intro-to-writing-data.md).
+[Intro to Writing Data](./intro-to-writing-data).
```
[the remark this answers is not in the document]
Comment (mikemaccana) [reply content/courses/intro-to-solana/intro-to-custom-onchain-programs.md]: Done

File: content/courses/mint-extensions/metadata.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+author: unboxed
+title: Token Extensions for Mints
```
[the remark this answers is not in the document]
Comment (mikemaccana) [reply content/courses/mint-extensions/metadata.yml:2]: 100% that's way better.

**Comment by mikemaccana** [comment]
> update all relative links (i.e. ./file). they do not work in this repo ✅
> Mentioned Anchor in "Onchain program development" metadata ✅
> Added missing heading (to security-intro) plus some lesson -> course fixes  ✅
> Wrap the form link in a "success" Callout to make it stand out some more  ✅

**Review by nickfrosty — APPROVED** [review]

**Pull request merged**